### ISSUE TITLE
Problems with GeoTools version 20.2 when using MATSim 11.0

### ISSUE THREAD

**Issue by alexanderbachev** [matsim-org/matsim-code-examples#340]
Dear guys, 

I have posted a problem I have in the pt2matsim Issues section, but figured out it actually belongs here. I am trying to use the pt2matsim plug-in to create a transitSchedule file from a GTFS feed. It was pointed out to me by Dr. Rieser, that it works with MATSim 11.0, while I was using a 12.0-2019w48-SNAPSHOT. I've tried changing the version to 11.0 in the pom file, but when doing that, my project no longer builds and I get some errors concerning the geotools library:
```
Information:Too many modules require recompilation, forcing full project rebuild
Information:java: Errors occurred while compiling module 'matsim-example-project'
Information:javac 13.0.2 was used to compile java sources
Information:20.05.20, 14:58 - Build completed with 11 errors and 0 warnings in 2 s 624 ms
Error:java: error reading /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar; zip END header not found
Error:java: error reading /Users/alexanderbachev/.m2/repository/org/geotools/gt-referencing/20.2/gt-referencing-20.2.jar; zip END header not found
Error:java: error reading /Users/alexanderbachev/.m2/repository/org/geotools/gt-shapefile/20.2/gt-shapefile-20.2.jar; zip END header not found
Error:java: error reading /Users/alexanderbachev/.m2/repository/org/geotools/gt-geotiff/20.2/gt-geotiff-20.2.jar; zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RunCreatePopulationRandom.java
Error:(1, 1) java: cannot access org.matsim.project
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RUN.java
Error:(1, 1) java: cannot access unnamed package
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/gui/MATSimGUI.java
Error:(20, 1) java: cannot access org.matsim.gui
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RunCreateNetworkFromOSM.java
Error:(28, 8) java: cannot access java.lang
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
Error:(61, 51) java: cannot access OsmNetworkReader
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RunMatsimFromExamplesUtils.java
Error:(13, 1) java: cannot access org.matsim
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RunMatsim.java
Error:(31, 8) java: cannot access org
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
```
Does anybody have an insight into what is causing the problem? I've tried adding the version 20.2 of the above mentioned dependencies in the pom, but to no avail.

Thanks!

**Comment by alexanderbachev** [comment]
Can it be that what causes the problem is that the URL for the osgeo repository in pt2matsim is no longer valid? (The new one is: https://repo.osgeo.org/repository/release/)

**Comment by JWJoubert** [comment]
Most likely, @alexanderbachev that would be my suggestion. The new entry in the `pom.xml` file should be:
```
<repository>
    <!-- Geotools is not on Maven central -->
    <id>osgeo</id>
    <name>Geotools repository</name>
    <url>https://repo.osgeo.org/repository/release/</url>
    <snapshots><enabled>false</enabled></snapshots>
    <releases><enabled>true</enabled></releases>
</repository>
```

**Comment by alexanderbachev** [comment]
> Most likely, @alexanderbachev that would be my suggestion. The new entry in the `pom.xml` file should be:
> 
> ```
> <repository>
>     
>     <id>osgeo</id>
>     <name>Geotools repository</name>
>     <url>https://repo.osgeo.org/repository/release/</url>
>     <snapshots><enabled>false</enabled></snapshots>
>     <releases><enabled>true</enabled></releases>
> </repository>
> ```
@JWJoubert , I have changed that into the pom file of my project a month ago. The pom file of the pt2matsim plug-in is still not updated and contains the old URL. Unfortunately, I can't change that bit there, because the file is read-only. ( I am trying to access it from the External libraries section in my project.)

**Comment by alexanderbachev** [comment]
Hello, guys. After a fresh cloning of the example-project, which still couldn't build with version 11 of matsim, I think I figured out why version 11 of matsim is failing to build. I navigated through the directories in the repository server (https://dl.bintray.com/matsim/matsim/org/matsim/matsim/11.0/) and found that the pom file of matsim v11 is still having the old osgeo url, which is no longer working. Is it possible to change it? Otherwise I can't use the pt2matsim plug-in which only works with matsim v11. 

Is there a way to edit the pom files myself? I tried editing them from the external lib section in IntelliJ, but the files are read-only. Please, let me know, if there is a way to fix this.

**Comment by kainagel** [comment]
I thought that repository locations are (1) not inherited but (2) only set at the level of your current project.  In consequence, if you are cloning matsim-example-project, and the geotools repo is wrong in the pom of the project (which is, in fact, no longer the case), then correcting the repo location in that local pom file should be sufficient, even if this is wrong in upstream pom.xml.  Am I remembering this wrong, or did this change?

**Comment by alexanderbachev** [comment]
@kainagel , the osgeo repo has indeed been updated to the current url some weeks ago, which fixed the issue. The project builds without any problems, when the latest snapshot version of MATSim (12.0-2019w48-SNAPSHOT) is selected in the pom file. However, when I change the version to the (supposedly) stable release 11.0. The project fails to build, throwing the following errors:
```
Information:Too many modules require recompilation, forcing full project rebuild
Information:java: Errors occurred while compiling module 'matsim-example-project'
Information:javac 13.0.2 was used to compile java sources
Information:20.05.20, 14:58 - Build completed with 11 errors and 0 warnings in 2 s 624 ms
Error:java: error reading /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar; zip END header not found
Error:java: error reading /Users/alexanderbachev/.m2/repository/org/geotools/gt-referencing/20.2/gt-referencing-20.2.jar; zip END header not found
Error:java: error reading /Users/alexanderbachev/.m2/repository/org/geotools/gt-shapefile/20.2/gt-shapefile-20.2.jar; zip END header not found
Error:java: error reading /Users/alexanderbachev/.m2/repository/org/geotools/gt-geotiff/20.2/gt-geotiff-20.2.jar; zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RunCreatePopulationRandom.java
Error:(1, 1) java: cannot access org.matsim.project
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RUN.java
Error:(1, 1) java: cannot access unnamed package
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/gui/MATSimGUI.java
Error:(20, 1) java: cannot access org.matsim.gui
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RunCreateNetworkFromOSM.java
Error:(28, 8) java: cannot access java.lang
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
Error:(61, 51) java: cannot access OsmNetworkReader
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RunMatsimFromExamplesUtils.java
Error:(13, 1) java: cannot access org.matsim
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
/Users/alexanderbachev/matsim-example-project/src/main/java/org/matsim/project/RunMatsim.java
Error:(31, 8) java: cannot access org
  Error reading file /Users/alexanderbachev/.m2/repository/org/geotools/gt-main/20.2/gt-main-20.2.jar: zip END header not found
```
That happens even when I freshly clone the matsim-example-project. As I've said earlier, I have almost no experience with maven, I am just assuming that the issue could be related to the same url change, as all errors concern geotools. It would be perhaps helpful if someone could try to build the project with MATSim version 11.0 to see if the issue is just in my system or everywhere.

**Comment by kainagel** [comment]
Maybe maven repository corruption?  Can you move or rename or remove `/Users/alexanderbachev/.m2/repository/org/geotools` (the whole directory), try again, and report?

**Comment by alexanderbachev** [comment]
Just before I received your suggestion, I had deleted the hole .m2/repository directory and reimported the whole maven project anew. I know longer get the above mentioned errors. 
The only error I get know is:
```
/Users/alexanderbachev/matsim-example-project/src/test/java/org/matsim/project/RunMatsimTest.java
Error:(49, 55) java: cannot find symbol
  symbol:   method extendUrl(java.net.URL,java.lang.String)
  location: class org.matsim.core.utils.io.IOUtils
```
I have unmarked the src/test/java directory as Test Source directory, which temporarily resolved the problem and am now able to build the project. 

It looks like repository corruption was the issue, indeed. Thank you for your help!
